tests/utils.py

```diff
@@ -438,3 +439,19 @@ def scriptpubkey_addr(scriptpubkey):
         # Modern bitcoin (at least, git master)
         return scriptpubkey['address']
     return None
+
+
+def trace_pay_command(rpc_pay_call, node):
+    """Utility function that run the function and prints information
+    about the channels, and payment status to catch some unrelated error
+    otherwise return the result of the lambda function.
+    """
+    try:
+        return rpc_pay_call()
+    except RpcError as rpc_err:
+        # Enable the trace with --log-cli-level=INFO
+        # and void to put other info inside the normal console log.
+        logging.info(node.rpc.listfunds())
+        logging.info(node.rpc.paystatus())
+        logging.info(node.rpc.listpeerchannels())
+        raise rpc_err
```

Comment:
If you run pytest as we do in GH CI, as "TEST_DEBUG=1 pytest -p no:logging" you get all the rpc calls and their results already!

Reply:
Oh right, this was a hack to trace the pay command, with the logging lib we can have the log print in the terminal and it should be easy to trace in the terminal.

Is there a better way to implement this trace function? I think this will be very useful during the developing lifecycle but I hope we do not need inside the CI when the test are stable